src/Components/Form/Form.js

```diff
@@ -1,91 +1,119 @@
 import './Form.scss';
 
 export class Form extends Component {
   constructor(props) {
     super(props)
     this.state = {
+      prompts: [],
       allAnswers: [],
       currentAnswers: [],
       questionsPerActivity: [],
     }
   }
 
+  componentDidMount = async () => {
+    this.setState({prompts: await getAllQuestions()})
+  }
 
-  updateCurrentAnswers = (event) => {
-    event.preventDefault();
-    if (this.state.currentAnswers.includes(event.target.textContent)){
-      return 
+  updateCurrentAnswers = (event, questionData) => {
+    if (+questionData.allowedResponses && this.state.currentAnswers.find(answer => questionData.choices.includes(answer))) {
+      let choiceToSwap = this.state.currentAnswers.find(answer => questionData.choices.includes(answer))
+      let indexToReplace = this.state.currentAnswers.indexOf(choiceToSwap)
+      let answerToReplace = this.state.currentAnswers.slice(indexToReplace + 1)
+      if (answerToReplace) {
+        return this.setState({currentAnswers: [...answerToReplace, event.target.textContent]})
+      }
+    }
+    else {
+      this.setState({currentAnswers: [...this.state.currentAnswers, event.target.textContent]})
     }
     this.props.updateActivityAnswers(event)
-    this.setState({currentAnswers: [...this.state.currentAnswers, event.target.textContent]})
   }
 
   updateAllAnswers = async (event) => {
     event.preventDefault();
     await this.setState({allAnswers: [...this.state.allAnswers, this.state.currentAnswers], currentAnswers: []})
     if (this.state.allAnswers.length === 1) {
       this.props.setActivities(this.state.allAnswers[0]);
-      let relevantQuestions = this.state.allAnswers[0].reduce((relevantQuestions, activity) => {
-        let filteredQuestions = questionSet.filter(question => {
-          return question.activity === activity; 
-        })
-        let questionsByActivity = {
-          activity: activity,
-          questions: filteredQuestions
-        }
-        relevantQuestions.push(questionsByActivity)
-        return relevantQuestions;  
-      }, [])
+      let activitySet = this.handleBothGames()
+      let relevantQuestions = this.determineRelevantQuestions(activitySet)
       this.setState({questionsPerActivity: [...relevantQuestions]})
     }
- }
+  }
+  determineRelevantQuestions = (activitySet) => {
+    return activitySet.reduce((relevantQuestions, activity) => {
+      let filteredQuestions = this.state.prompts.filter(question => {
+        return question.activity === activity;
+      })
+      let questionsByActivity = {
+        activity: activity,
+        questions: filteredQuestions
+      }
+      relevantQuestions.push(questionsByActivity)
+      return relevantQuestions;
+    }, [])
+  }
+
+  handleBothGames = () => {
+    if (this.state.allAnswers[0].includes('card games') && this.state.allAnswers[0].includes('board games')) {
+      let filteredSet = this.state.allAnswers[0].sort((a, b) => {
+        return a > b ? - 1 : 1
```

Comment:
It was interesting to talk through what's happening on line 62.  Will keep in mind that this syntax is helpful when trying to sort letters rather than numbers.